src/index/IndexImpl.Text.cpp

```diff
@@ -246,28 +239,43 @@ void IndexImpl::addTextFromOnDiskIndex() {
 }
 
 // _____________________________________________________________________________
-size_t IndexImpl::processWordsForVocabulary(string const& contextFile,
-                                            bool addWordsFromLiterals) {
+size_t IndexImpl::processWordsForVocabulary(const string& file,
+                                            bool addWordsFromLiterals,
+                                            bool useDocsFileForVocabulary) {
   size_t numLines = 0;
   ad_utility::HashSet<string> distinctWords;
-  for (auto line : wordsInTextRecords(contextFile, addWordsFromLiterals)) {
+  auto processLine = [&numLines, &distinctWords](const WordsFileLine& line) {
     ++numLines;
-    // LOG(INFO) << "LINE: "
-    //           << std::setw(50) << line.word_ << "   "
-    //           << line.isEntity_ << "\t"
-    //           << line.contextId_.get() << "\t"
-    //           << line.score_ << std::endl;
     if (!line.isEntity_) {
       distinctWords.insert(line.word_);
     }
+  };
+  if (useDocsFileForVocabulary) {
+    for (const auto& line :
+         getWordsLineFromDocsFile(file, textVocab_.getLocaleManager())) {
+      processLine(line);
+    }
+  } else {
+    for (const auto& line :
+         WordsFileParser{file, textVocab_.getLocaleManager()}) {
+      processLine(line);
+    }
+  }
```

Comment:
```suggestion
const auto& localeManager = textVocab_.getLocaleManager();
  if (useDocsFileForVocabulary) {
    ql::ranges::for_each(getWordsLineFrom(file, localManager), processLine;
    }
  } else {
    for (const auto& line :
         WordsFileParser{file, textVocab_.getLocaleManager()}) {
      processLine(line);
    }
  }
```
(same refactoring for the otther `for(...){processLine()` loops, saves two lines and increases readability).

Reply:
1620441c80164b4545bea8aace49fdf91dbfc8da